src/core/control/Control.cpp

```diff
@@ -1979,7 +1992,35 @@ void Control::toolLineStyleChanged() {
     const LineStyle& lineStyle = toolHandler->getTool(TOOL_PEN).getLineStyle();
     string style = StrokeStyle::formatStyle(lineStyle);
 
-    if (style == "dash") {
+    bool isLineStyleSameForAll = true;
+	
+    if (win) {
+        EditSelection* sel = win->getXournal()->getSelection();
+        if (sel) {
+			bool isFirstStrokeElement = true;
+			string previous_style = "none";
+			
```

Comment:
Just as a reminder, for when we pass to c++ 20

```suggestion
			
	\\ Todo(cpp20) Replace with std::ranges::filter_view and for_first_then_for_each
```

Reply:
Added in a1f5563482acc2a3b6becf0801ac0b6a92f22363.

Sorry, due to the refactoring I'm not entirely sure if I put it in the correct spot.

Can you quickly check if it's correct?

Reply:
Yes, that's the right place